clients/unshaded/src/main/java/tachyon/client/file/BaseFileSystem.java

```diff
@@ -0,0 +1,276 @@
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package tachyon.client.file;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import tachyon.TachyonURI;
+import tachyon.annotation.PublicApi;
+import tachyon.client.file.options.CreateDirectoryOptions;
+import tachyon.client.file.options.CreateFileOptions;
+import tachyon.client.file.options.DeleteOptions;
+import tachyon.client.file.options.ExistsOptions;
+import tachyon.client.file.options.FreeOptions;
+import tachyon.client.file.options.GetStatusOptions;
+import tachyon.client.file.options.ListStatusOptions;
+import tachyon.client.file.options.LoadMetadataOptions;
+import tachyon.client.file.options.MountOptions;
+import tachyon.client.file.options.OpenFileOptions;
+import tachyon.client.file.options.RenameOptions;
+import tachyon.client.file.options.SetAttributeOptions;
+import tachyon.client.file.options.UnmountOptions;
+import tachyon.exception.DirectoryNotEmptyException;
+import tachyon.exception.FileAlreadyExistsException;
+import tachyon.exception.FileDoesNotExistException;
+import tachyon.exception.InvalidPathException;
+import tachyon.exception.TachyonException;
+import tachyon.thrift.FileInfo;
+
+/**
+* Default implementation of the {@link FileSystem} interface. Developers can extend this class
+* instead of implementing the interface. This implementation reads and writes data through
+* {@link FileOutStream} and {@link FileInStream}.
+*/
+@PublicApi
+public class BaseFileSystem implements FileSystem {
+  @Override
+  public TachyonURI createDirectory(TachyonURI path)
+      throws FileAlreadyExistsException, InvalidPathException, IOException, TachyonException {
+    return createDirectory(path, CreateDirectoryOptions.defaults());
+  }
+
+  @Override
+  public TachyonURI createDirectory(TachyonURI path, CreateDirectoryOptions options)
+      throws FileAlreadyExistsException, InvalidPathException, IOException, TachyonException {
+    FileSystemMasterClient masterClient = FileSystemContext.INSTANCE.acquireMasterClient();
+    try {
+      return masterClient.createDirectory(path, options);
+    } finally {
+      FileSystemContext.INSTANCE.releaseMasterClient(masterClient);
+    }
+  }
+
+  @Override
+  public FileOutStream createFile(TachyonURI path)
```

Comment:
I assume you'll rename `TachyonURI` to `URI` in a subsequent PR for consistency?